app/models/document.rb

```diff
@@ -37,21 +37,31 @@ def validate_parameters(content, mimetype)
     avm_service.validate_parameters(self, content, mimetype)
   end
 
+  def signature_validation
+    avm_service.validate self
+  end
+
   def signers
-    [
+    response = signature_validation
+
+    return [] if !response['signedObjects'] || response['signedObjects'].size != 1 || (response['unsignedObjects'] && response['unsignedObjects'].size != 0)
+    object_id = response['signedObjects'][0]['id']
+
+    response['signatures'].select{ |s| s['signedObjectsIds'].include? object_id }.map do |signature|
       {
-        "signedBy": "SERIALNUMBER=PNOSK-1234567890, C=SK, L=Bratislava, SURNAME=Smith, GIVENNAME=John, CN=John Smith",
-        "issuedBy": "CN=SVK eID ACA2, O=Disig a.s., OID.2.5.4.97=NTRSK-12345678, L=Bratislava, C=SK"
+        signedBy: signature['signingCertificate']['subjectDN'],
+        issuedBy: signature['signingCertificate']['issuerDN']
       }
-    ]
+    end
   end
 
   def visualization
     avm_service.visualization(self)
   end
 
   def set_add_timestamp
-    parameters['level'] = parameters['level'].gsub(/BASELINE_B/, 'BASELINE_T')
+    parameters['level'] = parameters['level'].gsub(/BASELINE_B/, 'BASELINE_T') if parameters['level']
+    parameters['level'] = 'T' unless parameters['level']
```

Comment:
Toto neviem uplne co robi.

Reply:
2 veci:
1. pri chýbajúcom leveli by táto metóda padla na nil exception na volaní gsub.
2. Ak tam level nie je, chceme nejako povedať, že to chceme s pečiatkou. Preto vznikli krátke tvary SignatureLevel bez SignatureForm, čiže iba `B` alebo `T`. To podporuje aj AVM service, ktorý vie, že má/nemá pridať pečiatku, ale formu určí podľa už podpísaného dokumentu.